app/frontend/components/emoji_picker.js

```diff
@@ -0,0 +1,189 @@
+/* This file is part of Ezra Bible App.
+
+   Copyright (C) 2019 - 2021 Ezra Bible App Development Team <contact@ezrabibleapp.net>
+
+   Ezra Bible App is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   Ezra Bible App is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Ezra Bible App. See the file LICENSE.
+   If not, see <http://www.gnu.org/licenses/>. */
+
+const i18n = require('i18next');
+
+const template = html`
+  <style>
+    :host {
+      position: absolute; 
+      right: 0;
+      bottom: auto;
+      width: 1em;
+      margin-right: 1.2em;
+      padding: 3px;
+      fill: #5f5f5f;
+      display: inline-block;
+    }
+  </style>
+  <svg viewBox="0 0 496 512"><!-- Font Awesome Free 5.15.3 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) --><path d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm105.6-151.4c-25.9 8.3-64.4 13.1-105.6 13.1s-79.6-4.8-105.6-13.1c-9.9-3.1-19.4 5.4-17.7 15.3 7.9 47.1 71.3 80 123.3 80s115.3-32.9 123.3-80c1.6-9.8-7.7-18.4-17.7-15.3zM168 240c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160 0c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32z"/></svg>
+`;
+/**
+ * The emoji-picker component adds posibility to insert emojis to the tag names and notes 
+ * on desktop platforms (electron app). It provides emoji picker trigger button and inserts
+ * emoji in the previous DOM text input or codeMirror editor.
+ * 
+ * Emoji Picker dialog is implemented via external library:
+ * https://github.com/zhuiks/emoji-button
+ * That is customized version of original:
+ * https://github.com/joeattardi/emoji-button
+ * 
+ * Library instance is kept inside module variable picker.
+ * 
+ * @category Component
+ */
+class EmojiTrigger extends HTMLElement {
```

Comment:
The tag name `<emoji-picker>` was chosen because it's the most natural understanding for component user.
`EmojiTrigger` is a description of what the custom component is doing. It's not a picker itself, it just triggers the external picker.

But you are right it should be the same name. Otherwise it's a bit confusing. Maybe I'll rename it as `<emoji-button>` and `EmojiButton`. That would be logical as it's just a wrapper around external Emoji-Button library

Reply:
Whether or not it triggers the "external picker" ... at the end of the day it is a wrapper for the emoji button (that's how it behaves ...). I personally don't think the element name must be that technical. However, I leave it to you! :)